src/transaction_signature_options/mod.rs

```diff
@@ -119,95 +119,149 @@ impl interactive_clap::FromCli for Submit {
+                        Err(report) => {
+                            return interactive_clap::ResultFromCli::Err(
+                                optional_clap_variant,
+                                color_eyre::Report::msg(report),
+                            )
+                        }
                     }
-                };
-                eprintln!("{storage_message}");
-                interactive_clap::ResultFromCli::Ok(CliSubmit::Send)
-            }
+                    eprintln!("{storage_message}");
+                    interactive_clap::ResultFromCli::Ok(CliSubmit::Send)
+                }
+            },
             Some(CliSubmit::Display) => {
-                match (context.on_before_sending_transaction_callback)(
-                    &context.signed_transaction,
-                    &context.network_config,
-                    &mut storage_message,
-                ) {
-                    Ok(_) => (),
-                    Err(report) => {
-                        return interactive_clap::ResultFromCli::Err(
-                            optional_clap_variant,
-                            color_eyre::Report::msg(report),
-                        )
+                match context.signed_transaction_or_signed_delegate_action {
+                    SignedTransactionOrSignedDelegateAction::SignedTransaction(
+                        signed_transaction,
+                    ) => {
+                        if let Err(report) = (context.on_before_sending_transaction_callback)(
+                            &signed_transaction,
+                            &context.network_config,
+                            &mut storage_message,
+                        ) {
+                            return interactive_clap::ResultFromCli::Err(
+                                optional_clap_variant,
+                                color_eyre::Report::msg(report),
+                            );
+                        };
+                        let base64_transaction = near_primitives::serialize::to_base64(
+                            signed_transaction
+                                .try_to_vec()
+                                .expect("Transaction is not expected to fail on serialization"),
+                        );
+                        eprintln!("\nSerialize_to_base64:\n{}", &base64_transaction);
+                        eprintln!("{storage_message}");
+                        interactive_clap::ResultFromCli::Ok(CliSubmit::Display)
+                    }
+                    SignedTransactionOrSignedDelegateAction::SignedDelegateAction(
+                        signed_delegate_action,
+                    ) => {
+                        let base64_transaction = near_primitives::serialize::to_base64(
+                            signed_delegate_action
+                                .try_to_vec()
+                                .expect("Transaction is not expected to fail on serialization"),
+                        );
```

Comment:
Construct the relevant SignedTransactionAsBase64 or SignedDelegateActionAsBase64 and then just eprint it (Display trait will do the job).

Reply:
Done